llvm/lib/Object/Archive.cpp

```diff
@@ -969,12 +969,19 @@ Archive::Archive(MemoryBufferRef Source, Error &Err)
   Err = Error::success();
 }
 
+object::Archive::Kind Archive::getDefaultKindForTriple(Triple &T) {
+  if (T.isOSDarwin())
+    return object::Archive::K_DARWIN;
+  if (T.isOSAIX())
+    return object::Archive::K_AIXBIG;
+  if (T.isOSWindows())
+    return object::Archive::K_COFF;
```

Comment:
I'm a little nervous about how this impacts the default archive format on a Windows host: our downstream uses a cross-compiler, typically hosted on Windows, but using GNU-style archives (and the objects therefore have a different triple too).

I'm going to suggest to our internal binutils team that they run some testing with this patch applied, to see if it makes any difference.

Reply:
Please note the host matters only when `llvm-ar` can't infer the format from any other source. In practice, it means that it matters only when creating empty archives and archives who's first member is a non-symbolic files. In typical use case, when you pass object files, this PR shouldn't change anything unless you use COFF files.

I could limit impact of this PR by not changing the effect of `getDefaultKindForHost`, but it feels right to be consistent. If we consider those cases to be a problem, then it's also problematic for existing `isOSDarwin()` and `isOSAIX()` checks (which have even more significant format differences). If user cares about the format and uses `llvm-ar ` on non-symbolic files, then an explicit `--format` argument is the only reliable way I can see to handle all cross compilation variants.

Reply:
Thanks @jh7370, I checked and this will change our downstream ar's output in the cases @cjacek highlights above. However, due to @cjacek's points I cannot argue against it's submission.

The current behavior that default output format is defined by host does not seem good for the cross-compile use case in general. It may be preferable for the default to be based on a CMAKE option or something similar to gnu binutils and the GNUTARGET environment variable. Both are likely out of the scope of this PR though.

Reply:
LLVM_DEFAULT_TARGET_TRIPLE sounds like a good idea, I will create a separated PR for it, thanks.